### PR TITLE
Add ephyviewer recipe

### PR DESCRIPTION
**Pull request by jpgill86**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by jpgill86** [comment]
@samuelgarcia, I have listed you as a co-maintainer for this conda-forge recipe. If that is OK with you, could you post a comment here saying so? This will satisfy the last item on my checklist.

**Comment by jpgill86** [comment]
Hmm, import test after build is failing on Linux (but not Windows or Mac) due to a problem with PyQt.

```
import: 'ephyviewer'
no PyQt5/PyQt4
Traceback (most recent call last):
  File "/home/conda/staged-recipes/build_artifacts/ephyviewer_1573258631532/test_tmp/run_test.py", line 2, in <module>
    import ephyviewer
  File "/home/conda/staged-recipes/build_artifacts/ephyviewer_1573258631532/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.7/site-packages/ephyviewer/__init__.py", line 8, in <module>
    from .mainviewer import MainViewer, compose_mainviewer_from_sources
  File "/home/conda/staged-recipes/build_artifacts/ephyviewer_1573258631532/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.7/site-packages/ephyviewer/mainviewer.py", line 10, in <module>
    from .navigation import NavigationToolBar
  File "/home/conda/staged-recipes/build_artifacts/ephyviewer_1573258631532/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.7/site-packages/ephyviewer/navigation.py", line 5, in <module>
    import pyqtgraph as pg
  File "/home/conda/staged-recipes/build_artifacts/ephyviewer_1573258631532/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.7/site-packages/pyqtgraph/__init__.py", line 13, in <module>
    from .Qt import QtGui
  File "/home/conda/staged-recipes/build_artifacts/ephyviewer_1573258631532/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.7/site-packages/pyqtgraph/Qt.py", line 153, in <module>
    from PyQt5 import QtGui, QtCore, QtWidgets, uic
ImportError: libGL.so.1: cannot open shared object file: No such file or directory
Tests failed for ephyviewer-1.2.0-py_0.tar.bz2 - moving package to /home/conda/staged-recipes/build_artifacts/broken
WARNING conda_build.build:tests_failed(2231): Tests failed for ephyviewer-1.2.0-py_0.tar.bz2 - moving package to /home/conda/staged-recipes/build_artifacts/broken
TESTS FAILED: ephyviewer-1.2.0-py_0.tar.bz2
##[error]Bash exited with code '1'.
```

PyQt is not listed in the recipe as an explicit requirement of ephyviewer because that dependency is established indirectly through pyqtgraph. This is working for Windows and Mac, and I [can see](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=91654&view=logs&jobId=10bda42a-d420-5c28-a43f-bc4e66405e5c&taskId=bef775d0-a1a2-55a3-9721-17147029baf0&lineStart=749&lineEnd=753&colStart=5&colEnd=57) that the Linux build installs `pyqt` and `qt` too, yet it fails nevertheless. Puzzling...

**Comment by jan-janssen** [comment]
Can you try https://conda-forge.org/docs/maintainer/knowledge_base.html#libgl ? 

```
requirements:
  build:
    - {{ cdt('mesa-libgl-devel') }}  # [linux]
    - {{ cdt('mesa-dri-drivers') }}  # [linux]
    - {{ cdt('libselinux') }}  # [linux]
    - {{ cdt('libxdamage') }}  # [linux]
    - {{ cdt('libxxf86vm') }}  # [linux]
  host:
    - xorg-libxfixes  # [linux]
```

**Comment by jpgill86** [comment]
Thank you @jan-janssen for helping! I didn't see the section about libGL!

I'm going to first revert the addition of ``pyqt`` to the run requirements, since that changed nothing.

I think I've read that ``noarch: python`` packages can't use selectors like ``[linux]``. Does this mean I should drop the ``noarch: python`` too? I will try it first without ``noarch: python``, and we can put it back in if you think it's possible.

ephyviewer is a pure Python package. Only our dependency (pyqtgraph) needs libGL, and I saw that [pyqtgraph's recipe](https://github.com/conda-forge/pyqtgraph-feedstock/blob/dfaaaf029d8d1bd4d9f13a6b565227d50374a5c2/recipe/meta.yaml) doesn't include any special build instructions for linking libGL. That recipe is over a year old, so maybe the conda build environments have changed. It's worth a shot!

**Comment by jpgill86** [comment]
Hmm, now builds are failing across all platforms.

For Linux, the error is different this time and doesn't complain about libGL, though it still says "no PyQt5/PyQt4" (which is a warning printed by ephyviewer when PyQT cannot be found).

```
import: 'ephyviewer'
no PyQt5/PyQt4
Traceback (most recent call last):
  File "/home/conda/staged-recipes/build_artifacts/ephyviewer_1573316532367/test_tmp/run_test.py", line 2, in <module>
    import ephyviewer
  File "/home/conda/staged-recipes/build_artifacts/ephyviewer_1573316532367/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python2.7/site-packages/ephyviewer/__init__.py", line 6, in <module>
    from .icons import *
  File "/home/conda/staged-recipes/build_artifacts/ephyviewer_1573316532367/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python2.7/site-packages/ephyviewer/icons/__init__.py", line 6, in <module>
    from . import icons
ImportError: cannot import name icons
Tests failed for ephyviewer-1.2.0-py27_0.tar.bz2 - moving package to /home/conda/staged-recipes/build_artifacts/broken
WARNING conda_build.build:tests_failed(2231): Tests failed for ephyviewer-1.2.0-py27_0.tar.bz2 - moving package to /home/conda/staged-recipes/build_artifacts/broken
TESTS FAILED: ephyviewer-1.2.0-py27_0.tar.bz2
##[error]Bash exited with code '1'.
```

The Windows and Mac builds both do NOT complain about "no PyQt5/PyQt4", but they also get stuck with ``ImportError: cannot import name icons``.

I believe this is a Python 2 vs 3 issue. ephyviewer has partial support for Python 2 that is currently being improved (NeuralEnsemble/ephyviewer#80). I can see that the build and/or test is using Python 2.7, which is causing [this code](https://github.com/NeuralEnsemble/ephyviewer/blob/6f78e3c1703a41feaf35085ece759b564609834b/ephyviewer/icons/__init__.py#L6) to run, which actually contains a bug in the released version. Since the fix in the PR isn't included in the released version I'm going to restrict the conda-forge recipe to Python 3 only and try again.

---

By the way, it's actually PyQT, not pyqtgraph (which depends on PyQT), that directly needs libGL at build time. I can see that [PyQT's recipe](https://github.com/conda-forge/pyqt-feedstock/blob/58404009beea612eaa730e33ce0bbfb2d09048cd/recipe/meta.yaml) contains most -- but not all -- of the requirements listed on that [Knowledge Base page](https://conda-forge.org/docs/maintainer/knowledge_base.html#libgl).

**Comment by jpgill86** [comment]
The addition of ``python >=3.4`` to the build and run requirements has created package conflicts, which differ between platforms.

LINUX
```
Package python conflicts for:
scipy -> numpy[version='1.11.*,1.12.*,>=1.9.3,<2.0a0'] -> python[version='2.7.*,3.5.*,3.6.*,>=2.7,<2.8.0a0,>=3.5,<3.6.0a0,>=3.6,<3.7.0a0']
pyqtgraph[version='>=0.10.0'] -> pyqt=4.11 -> qt[version='4.8.*,5.9.*,>=5.6.2,<5.7.0a0,>=5.9.6,<5.10.0a0,>=5.9.7,<5.10.0a0'] -> gtk2 -> gdk-pixbuf -> gobject-introspection -> python[version='>=2.7,<2.8.0a0,>=3.7,<3.8.0a0,>=3.8,<3.9.0a0']
matplotlib[version='>=2.0'] -> pyqt[version='5.*,5.6.*,5.9.*,>=5.9.2,<5.10.0a0'] -> qt[version='4.8.*,5.9.*,>=5.6.2,<5.7.0a0,>=5.9.6,<5.10.0a0,>=5.9.7,<5.10.0a0'] -> gtk2 -> gdk-pixbuf -> gobject-introspection -> python[version='>=2.7,<2.8.0a0,>=3.7,<3.8.0a0,>=3.8,<3.9.0a0']
numpy -> python[version='2.7.*,3.5.*,3.6.*,>=2.7,<2.8.0a0,>=3.5,<3.6.0a0,>=3.6,<3.7.0a0']
python[version='>=3.8,<3.9.0a0']
```

MAC
```
Package python conflicts for:
numpy -> mkl_random -> mkl-service[version='>=2,<3.0a0'] -> six -> python[version='2.7.*,>=3.5,<3.6.0a0,>=3.6,<3.7.0a0,>=3.7,<3.8.0a0']
scipy -> numpy[version='1.10.*,1.11.*,>=1.11.3,<2.0a0,>=1.9,>=1.9.3,<2.0a0'] -> mkl_random -> mkl-service[version='>=2,<3.0a0'] -> six -> python[version='2.7.*,>=3.5,<3.6.0a0,>=3.6,<3.7.0a0,>=3.7,<3.8.0a0']
matplotlib[version='>=2.0'] -> numpy -> mkl_random -> mkl-service[version='>=2,<3.0a0'] -> six -> python[version='2.7.*,>=3.5,<3.6.0a0,>=3.6,<3.7.0a0,>=3.7,<3.8.0a0']
python[version='>=3.8,<3.9.0a0']
pyqtgraph[version='>=0.10.0'] -> numpy -> mkl_random -> mkl-service[version='>=2,<3.0a0'] -> six -> python[version='2.7.*,>=3.5,<3.6.0a0,>=3.6,<3.7.0a0,>=3.7,<3.8.0a0']
```

WINDOWS
```
Package python conflicts for:
python[version='>=3.8,<3.9.0a0']
pyqtgraph[version='>=0.10.0'] -> numpy -> mkl_random[version='>=1.0.2,<2.0a0'] -> mkl-service[version='>=2,<3.0a0'] -> six -> python[version='>=2.7,<2.8.0a0,>=3.6,<3.7.0a0,>=3.7,<3.8.0a0']
matplotlib[version='>=2.0'] -> numpy -> mkl_random[version='>=1.0.2,<2.0a0'] -> mkl-service[version='>=2,<3.0a0'] -> six -> python[version='>=2.7,<2.8.0a0,>=3.6,<3.7.0a0,>=3.7,<3.8.0a0']
scipy -> numpy[version='>=1.11.3,<2.0a0,>=1.15.1,<2.0a0'] -> mkl_random[version='>=1.0.2,<2.0a0'] -> mkl-service[version='>=2,<3.0a0'] -> six -> python[version='>=2.7,<2.8.0a0,>=3.6,<3.7.0a0,>=3.7,<3.8.0a0']
numpy -> mkl_random[version='>=1.0.2,<2.0a0'] -> mkl-service[version='>=2,<3.0a0'] -> six -> python[version='>=2.7,<2.8.0a0,>=3.6,<3.7.0a0,>=3.7,<3.8.0a0']
```

**Comment by jpgill86** [comment]
From the package conflict errors, I can see that numpy is imposing Python<3.8. But I don't see why that's a problem, other than the fact that "``python[version='>=3.8,<3.9.0a0']``" appears among the error messages. I don't know where this is coming from. None of the other requirements (scipy, matplotlib, pyqtgraph) impose this version restriction.

At this point, I am at a loss and could really use some outside help (@jan-janssen?). I don't understand why our little pure Python package would need all the libGL cruft that larger projects like pyqtgraph don't need, nor do I understand the package conflict issue that's currently plaguing the builds.

**Comment by jan-janssen** [comment]
I guess there is a issue with the current pyqtgraph recipe, as the modern builds fail https://github.com/conda-forge/pyqtgraph-feedstock/pull/8 so my recommendation would be to first get it to work, adding the necessary dependencies as I described above and focus on optimizing afterwards.

**Comment by jpgill86** [comment]
I see, that is... very frustrating. Thank you @jan-janssen for the quick reply!

**Comment by marcelotrevisani** [comment]
> I see, that is... very frustrating. Thank you @jan-janssen for the quick reply!

Try again in ~30 min please :)
Sorry for the delay

**Comment by jpgill86** [comment]
Thank you so much!

**Comment by jpgill86** [comment]
Now that pyqtgraph has a new build, I'm going to try reverting the libGL stuff to see what happens.

**Comment by jpgill86** [comment]
I'm still getting the same package conflict issues. 😭

**Comment by jpgill86** [comment]
Let's see what happens if I add back in ``noarch: python`` while retaining ``python >=3.4``. My understanding is that this is not allowed...

**Comment by jpgill86** [comment]
OK, maybe that's progress. Linux is still failing, again due to libGL stuff, and Windows and Mac are passing. We're basically back to where we started!

I'm going to follow the [pyqtgraph model](https://github.com/conda-forge/pyqtgraph-feedstock/pull/9#issuecomment-552126649) (thanks @marcelotrevisani!) and add a yum requirements file containing the libGL stuff. Let's see how that goes...

**Comment by jpgill86** [comment]
Amazing, tests are passing! To summarize:
- With ``noarch: python``, I just needed to add the ``yum_requirements.txt`` file containing ``mesa-libGL`` and ``libX11-devel``.
- I used that approach instead of the one outlined [here](https://github.com/conda-forge/staged-recipes/pull/10103#issuecomment-552107922) so that I wouldn't have to use the ``[linux]`` selectors.
- I wanted to avoid the ``[linux]`` selectors because they are (I think) incompatible with ``noarch: python``, and for some reason dropping ``noarch: python`` lead to package conflicts across all platforms (which I still don't understand).

Thank you @jan-janssen and @marcelotrevisani for your quick responses!

@samuelgarcia, I believe I just need your approval to include you among the maintainers for the recipe, then we'll be ready.

File: recipes/ephyviewer/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  noarch: python
+  number: 0
+  entry_points:
+    - ephyviewer=ephyviewer.scripts:launch_standalone_ephyviewer
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv "
```
Comment (marcelotrevisani) [inline recipes/ephyviewer/meta.yaml]:
```suggestion
  script: "{{ PYTHON }} -m pip install . -vv"
```

Comment (jpgill86) [reply recipes/ephyviewer/meta.yaml]: I can try that. I see that those flags are not present in ``example.yaml``, but I've also seen in the docs (e.g. [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#use-pip)) that ``--no-deps`` should be used. I think ``conda skeleton`` also put the flags in when I tried it. Was there a policy change, and the docs are out of date? Unfortunately [git blame](https://github.com/conda-forge/staged-recipes/blame/94aba5d772f68d7945453b68ffd32641bcb9e266/recipes/example/meta.yaml#L33) is not very useful for tracking down the reason for that change in ``example.yaml``.

**Comment by samuelgarcia** [comment]
Hi Jeffrey.
Thank you for this conda recipe.
This is OK for me to be listed in mainteners.

**Comment by jpgill86** [comment]
Thanks @samuelgarcia!

@jan-janssen, @marcelotrevisani: I believe this recipe is ready. Is it possible to merge?

**Comment by jpgill86** [comment]
Thank you @marcelotrevisani!

**Pull request merged**